### PR TITLE
Astra DB documentation updates, and ordering of indices

### PR DESCRIPTION
**Pull request by erichare**
# Description

This adds some needed documentation for the newly-merged Astra DB Vector Store integration. In the process, I also re-organized the various lists of vector stores alphabetically, which i think improves the readability of everything especially as more continue to be added.

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: docs/community/integrations/vector_stores.md [hunk 1]
```diff
@@ -86,59 +86,103 @@ response = query_engine.query("What did the author do growing up?")
...
-**Elasticsearch**
+**Apache Cassandra®**
 
-First, you can start Elasticsearch either locally or on [Elastic cloud](https://cloud.elastic.co/registration?utm_source=llama-index&utm_content=documentation).
+```python
+from llama_index.vector_stores import CassandraVectorStore
+import cassio
 
-To start Elasticsearch locally with docker, run the following command:
+# For an Astra DB cloud instance:
```
Comment (hemidactylus) [inline docs/community/integrations/vector_stores.md:95]: Perhaps `# To use an Astra DB cloud instance thrugh CQL:` or something?

File: docs/community/integrations/vector_stores.md [hunk 2]
```diff
@@ -86,59 +86,103 @@ response = query_engine.query("What did the author do growing up?")
...
-    # es_cloud_id="<cloud_id>",
-    # es_user="elastic",
-    # es_password="<password>",
+from llama_index.vector_stores import AstraDBVectorStore
+
+astra_db_store = AstraDBVectorStore(
+    token=os.environ["ASTRA_DB_APPLICATION_TOKEN"],  # Your Astra DB token
+    api_endpoint=os.environ[
+        "ASTRA_DB_API_ENDPOINT"
+    ],  # Your Astra DB API endpoint
```
Comment (hemidactylus) [inline docs/community/integrations/vector_stores.md:119]: Style formatting is a bit unfortunate and makes this seem "long-winded" (not our fault).
There is a dedicated notebook for the details, this is just to give the idea, so why not go with the simpler:
```
astra_db_store = AstraDBVectorStore(
    token="AstraCS:xY3b...",  # Your Astra DB token
    api_endpoint="https://012...abc-us-east1.apps.astra.datastax.com",  # Your Astra DB API endpoint
    collection_name="astra_v_table",  # Table name of your choice
    embedding_dimension=1536,  # Embedding dimension of the embeddings model used
)
```

File: docs/examples/vector_stores/AstraDBIndexDemo.ipynb [hunk 3]
```diff
@@ -0,0 +1,261 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Astra DB\n",
+    "\n",
+    ">[Astra DB](https://astra.datastax.com) is a NoSQL, row-oriented, highly scalable and highly available database.\n",
```
Comment (hemidactylus) [inline docs/examples/vector_stores/AstraDBIndexDemo.ipynb:9]: I'd link to https://docs.datastax.com/en/astra/home/astra.html

File: docs/examples/vector_stores/AstraDBIndexDemo.ipynb [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    "api_endpoint = input(\n",
+    "    \"\\nPlease enter your Database Endpoint URL (e.g. '0123abcd...'):\"\n",
+    ")\n",
+    "\n",
+    "token = getpass.getpass(\n",
+    "    \"\\nPlease enter your 'Database Administrator' Token (e.g. 'AstraCS:...'):\"\n",
+    ")\n",
+    "\n",
+    "OPENAI_API_KEY = getpass.getpass(\"OpenAI API Key:\")\n",
+    "openai.api_key = OPENAI_API_KEY"
```
Comment (hemidactylus) [inline docs/examples/vector_stores/AstraDBIndexDemo.ipynb:71]: If you want to shave one line off, you should be able to replace these two with `os.environ["OPENAI_API_KEY"] = getpass("OPENAI_API_KEY = ")`

File: docs/examples/vector_stores/AstraDBIndexDemo.ipynb (hunk 3 above)
Comment (hemidactylus) [reply docs/examples/vector_stores/AstraDBIndexDemo.ipynb:9]: This blurb was a mistake on my part. AI folks are not interested in these details. Suggestion (to your judgement):

`DataStax Astra DB is a serverless vector-capable database built on Cassandra and accessed through an easy-to-use JSON API.`

File: docs/community/integrations/vector_stores.md (hunk 2 above)
Comment (erichare) [reply docs/community/integrations/vector_stores.md:119]: I like that idea. really cleans up the appearance, and like you said, we have the notebook for env variable reading etc

**Comment by erichare** [comment]
@nerdai If you get a quick chance to take a look at this, i'd much appreciate it! Just tagging you because you reviewed our initial Astra DB integration :) this is just some documentation improvements related (and an additional sort of the respective integrations lists / table, which i thought would make things easier going forward)

**Comment by nerdai** [comment]
> @nerdai If you get a quick chance to take a look at this, i'd much appreciate it! Just tagging you because you reviewed our initial Astra DB integration :) this is just some documentation improvements related (and an additional sort of the respective integrations lists / table, which i thought would make things easier going forward)

Thanks for doing this! Taking a look at the new notebook now.

File: docs/examples/vector_stores/AstraDBIndexDemo.ipynb [hunk 5]
```diff
@@ -0,0 +1,209 @@
+{
```
Comment (nerdai) [inline docs/examples/vector_stores/AstraDBIndexDemo.ipynb:1]: <p>Can you add in a cell just above a snippet like the one below for an "Open In Colab" button?</p><p> </p><pre class="ql-syntax" spellcheck="false">&lt;a href="https://colab.research.google.com/github/run-llama/llama_index/blob/main/docs/examples/vector_stores/AsyncIndexCreationDemo.ipynb" target="_parent"&gt;&lt;img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/&gt;&lt;/a&gt;
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/8777/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

**Review by nerdai — APPROVED** [review]
Look's great @erichare! One small request for the notebook (i.e., add the Open In Colab function like in our other notebooks such as `AwadbDemo.ipynb`)

File: docs/examples/vector_stores/AstraDBIndexDemo.ipynb (hunk 5 above)
Comment (erichare) [reply docs/examples/vector_stores/AstraDBIndexDemo.ipynb:1]: @nerdai Good call, just added this :)

**Comment by erichare** [comment]
> Look's great @erichare! One small request for the notebook (i.e., add the Open In Colab function like in our other notebooks such as `AwadbDemo.ipynb`)

Thanks for looking at it @nerdai , hopefully good to go now!

**Comment by nerdai** [comment]
Looks great. Merging now. Thanks again!

**Pull request merged**